app/views/notifications/shared/_comment_box.html.erb

```diff
@@ -76,7 +76,7 @@
           <%= f.hidden_field :commentable_type, value: json_data["comment"]["commentable"]["class"]["name"] %>
 
           <%= f.hidden_field :parent_id, value: json_data["comment"]["id"] %>
-          <%= f.text_area :body_markdown, id: "comment-textarea-for-#{json_data['comment']['id']}", class: "crayons-textfield mb-2 resize-y", placeholder: "Reply...", 'aria-label': "Reply to a comment..." %>
+          <%= f.text_area :body_markdown, id: "comment-textarea-for-#{json_data['comment']['id']}", class: "crayons-textfield mb-2 resize-y", placeholder: "Reply...", 'aria-label': "Reply to a comment...", onfocus: "handleFocus(event)" %>
```

Comment:
Refactor (potentially blocking): We could attach these via event listeners on the client-side with a selector like `document.querySelectorAll('textarea[id^="comment-textarea-for-
"]')` instead of using an inline onfocus.

Reply:
I looked at doing that, but the issue I came up against is that the text areas aren't rendered initially, so we'd have to listen for DOM changes which felt a bit clunky, especially since we already have that chunk of code getting called when new comment text areas appear 🤔

Reply:
Suggestion (non-blocking): We could add a [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) to the container for the comments area to handle this so that it could pick up any new comment added. We can discuss further if you'd like, or perhaps this is an optimization for later.

Refactor (blocking): Add an `onblur` to remove the functionality so that it's only active when a community member is typing in the textarea. as we only ever want to have one active autocomplete combo box on the page.

Reply:
Looking into this now and remembering why I didn't add a blur event listener - essentially what happens at the moment is:

- text area is focused
- `<MentionAutocomplete />` is mounted, replacing any previous instance of an autocomplete
- User types `@` (or potentially other characters as per other requests here), `<MentionAutocompleteCombobox />` is mounted and the focus is transferred to an invisible combobox input. This triggers a 'blur' event on the original text area
- User selects an option/deletes all content including `@` sign/spaces or clicks away from search, and the `<MentionAutocompleteCombobox/>` is unmounted

Because I'm transferring focus to that invisible combobox input, the blur event is fired on the text area immediately as autocomplete search starts. Do you think what I've done in `base.jsx` is sufficient in making sure we never have two `MentionAutocomplete` instances?